### DISCUSSION TITLE
would you accept /flake.nix file PR so that ZN can be consumed by Nix right away?

### DISCUSSION THREAD

**Discussion by dzmitry-lahoda** [paritytech/zombienet#690] [Ideas]
flake.nix is NIX packaging format.

would it be maintained? 

1. there almost there no maintenance (it uses packag.json to build stuff)
2. any nix user is capable to patch it when issues arise
3. composable.finance is heavy on nix and migrating to ZN, so may be patches, if any, will be from there

**Comment by pepoviola** [comment]
Yes, totally.

**Comment by dzmitry-lahoda** [comment]
it works, but has issue in maintainability https://github.com/paritytech/zombienet/pull/772 (i create bug onto nixpkgs)

**Comment by dzmitry-lahoda** [comment]
seems people update nix hash even if it is not part of CI https://github.com/paritytech/zombienet/pull/886 . So i think will do another try to use nix builders which use package.lock integrity hashes (so nix will just validate what is npm does)

**Comment by dzmitry-lahoda** [comment]
I tried dream2nix. It uses hashes from lock, no need to up hash all the time. Did not worked. Because of loop inside package calling some heavy package stuff.

**Comment by dzmitry-lahoda** [comment]
```
       > npm ERR! Log files were not written due to an error writing to the directory: /nix/store/q4y11aq5lvp879pwlw6mvig6q7z6ly06-zombienet-npm-deps/_logs
```
this is hell. some deps writes to its own folder. seems will switch flake to releases.
